core-blocks/button/edit.js

```diff
@@ -106,22 +114,35 @@ class ButtonEdit extends Component {
 						keepPlaceholderOnFocus
 					/>
 					<InspectorControls>
-						<PanelColor
-							colorValue={ backgroundColor.value }
-							title={ __( 'Background Color' ) }
-							onChange={ setBackgroundColor }
-						/>
-						<PanelColor
-							colorValue={ textColor.value }
-							title={ __( 'Text Color' ) }
-							onChange={ setTextColor }
-						/>
-						{ this.nodeRef && <ContrastCheckerWithFallbackStyles
-							node={ this.nodeRef }
-							textColor={ textColor.value }
-							backgroundColor={ backgroundColor.value }
-							isLargeText={ true }
-						/> }
+						<PanelColorSettings
+							title={ __( 'Color Settings' ) }
+							colorSettings={ [
+								{
+									value: backgroundColor.value,
+									onChange: setBackgroundColor,
+									label: __( 'Background Color' ),
+									// translators: %s: The name of the color e.g: "vivid red" or color hex code if name is not available e.g: "#f00".
+									colorIndicatorAriaLabel: __( '(current background color: %s)' ),
```

Comment:
I would like to avoid the need for colorIndicatorAriaLabel, or at least make it optional. Can we derive the aria-label from the label, and maybe even remove current. e.g: use (background color: %s).

Reply:
How does that work with i18n? I thought the strings could only be statically analysed when using `__`.

Reply:
Maybe we can use something like: '(current %s: %s)' or '(%s: %s). Where we specify in the translator comments that the first string refers to the color label in lower case e.g: 'background color' and the second label to the color name or value.